### PR TITLE
Create AssetItemPopup

### PR DESCRIPTION
**Pull request by anngy12**
ProductPage is an example to show how it is used.
ProductDialog is the actual AssetItemPopup.

grafik

### CODE REVIEW

**Review by Sc3l3t0n — CHANGES_REQUESTED** [review]

**Comment by Sc3l3t0n** [comment]
Please rename the components to a more scoped name.

I would suggest adding `ShoppingCart` as a prefix.

File: src/TuDa.CIMS.Web/Components/ProductDialog.razor [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+                        </MudGrid>
+                    </MudCardContent>
+                </MudCard>
+            </MudItem>
+            <!-- Amount Input -->
+            <MudItem xs="12">
+                <MudTextField @bind-Value="Amount"
+                              Label="Amount"
+                              Min="1"
+                              Type="number"/>
```
[the remark this answers is not in the document]
Comment (anngy12) [reply src/TuDa.CIMS.Web/Components/ProductDialog.razor:30]: I got an error while using uint. But I found another good alternative to resolve the problem with negative numbers.
```suggestion
                <MudTextField @bind-Value="Amount"
                              Error="@isError"
                              ErrorText="Amount must be a positive number"
                              InputMode="@InputMode.numeric"
                              InputType="@InputType.Number"
                              Label="Amount"
                              T="int"/>
```

File: src/TuDa.CIMS.Web/Components/ProductPage.razor [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+            BindingSize = 0
+        };
+
+        var parameters = new DialogParameters
+        {
+            { "Product", product },
+            { "Amount", 1 } // Initial Value _amount
+        };
+
+        return DialogService.ShowAsync<ProductDialog>("Amount Input", parameters, options);
```
[the remark this answers is not in the document]
Comment (anngy12) [reply src/TuDa.CIMS.Web/Components/ProductPage.razor:42]: Console.WriteLine(await dialog.GetReturnValueAsync<uint>()); 
throws an error, when the dialog is closed through cancel

```suggestion
        return DialogService.ShowAsync<ProductDialog>("Amount Input", parameters, options);
```

File: src/TuDa.CIMS.Web/Components/ShoppingCartProductDialog.razor.cs [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    [Parameter]
+    public required AssetItem Product { get; set; }
+
+    //Purchase Entry to add to the List
+    public PurchaseEntry? PurchaseEntry { get; set; }
+    public static List<PurchaseEntry> Entries { get; set; } = [];
+
+    // Amount of Product
+    public int Amount { get; set; } = 1;
+    private bool isError => Amount <= 0;
```
Comment (Sc3l3t0n) [inline src/TuDa.CIMS.Web/Components/ShoppingCartProductDialog.razor.cs:25]: 🔧 Should be `IsError`

File: src/TuDa.CIMS.Web/Components/ShoppingCartProductDialog.razor.cs [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+namespace TuDa.CIMS.Web.Components;
+
+public partial class ShoppingCartProductDialog
+{
+    [CascadingParameter]
+    public required MudDialogInstance ProductDialog { get; set; }
+
+    [Inject]
+    public required ILogger<ShoppingCartProductDialog> Logger { get; set; }
```
Comment (Sc3l3t0n) [inline src/TuDa.CIMS.Web/Components/ShoppingCartProductDialog.razor.cs:13]: 🔧 Should be private

File: src/TuDa.CIMS.Web/Components/ShoppingCartProductDialog.razor.cs [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+            Logger.LogInformation($"Product added with quantity: {Amount}");
+
+            PurchaseEntry = new PurchaseEntry()
+            {
+                Amount = Amount,
+                AssetItem = Product,
+                PricePerItem = Product.Price,
+            };
+
+            Entries.Add(PurchaseEntry);
```
Comment (Sc3l3t0n) [inline src/TuDa.CIMS.Web/Components/ShoppingCartProductDialog.razor.cs:48]: 🔧 This stuff should be handled by the Page

File: src/TuDa.CIMS.Web/Components/ShoppingCartProductDialog.razor.cs [hunk 6]
```diff
@@ -10,19 +10,18 @@ public partial class ShoppingCartProductDialog
...
 
     // Product to be shown in Dialog.
     [Parameter]
     public required AssetItem Product { get; set; }
 
-    //Purchase Entry to add to the List
-    public PurchaseEntry? PurchaseEntry { get; set; }
-    public static List<PurchaseEntry> Entries { get; set; } = [];
+    [Parameter]
+    public EventCallback<int> OnAmountSubmitted { get; set; }
```
Comment (Sc3l3t0n) [inline src/TuDa.CIMS.Web/Components/ShoppingCartProductDialog.razor.cs:20]: 🔧 This is not needed because, that's returned when running Submit.

**Review by Sc3l3t0n — APPROVED** [review]

**Review by devidsdt — APPROVED** [review]

**Pull request merged**